Out-of-tree build broken

```
$ git clone https://github.com/JuliaLang/julia
$ cd julia
$ make O=build configure
$ make -j5
make[1]: *** No rule to make target 'jl_exports.h', needed by 'loader_exe.o'.  Stop.
make: *** [/home/tim/Julia/tmp/julia/Makefile:79: julia-cli-release] Error 2
```

Probable cause: https://github.com/JuliaLang/julia/pull/38160

--- a/cli/Makefile
+++ b/cli/Makefile
@@ -6,7 +6,7 @@ include $(JULIAHOME)/Make.inc
 include $(JULIAHOME)/deps/llvm-ver.make
 
 
-HEADERS := jl_exports.h loader.h $(addprefix $(JULIAHOME)/src/support/,platform.h dirpath.h) $(addprefix $(JULIAHOME)/src/,jl_exported_data.inc jl_exported_funcs.inc)
+HEADERS := $(addprefix $(JULIAHOME)/cli/,jl_exports.h loader.h) $(addprefix $(JULIAHOME)/src/support/,platform.h dirpath.h) $(addprefix $(JULIAHOME)/src/,jl_exported_data.inc jl_exported_funcs.inc)
 
 LOADER_CFLAGS = $(JCFLAGS) -I$(BUILDROOT)/src -I$(JULIAHOME)/src -I$(JULIAHOME)/src/support -I$(build_includedir) -ffreestanding
 LOADER_LDFLAGS = $(JLDFLAGS) -ffreestanding -L$(build_shlibdir) -L$(build_libdir)

